server/neptune/temporal/client.go

```diff
@@ -39,5 +73,154 @@ func NewClient(scope tally.Scope, logger logur.Logger, cfg valid.Temporal) (clie
 
 	}
 
-	return client.Dial(opts)
+	client, err := client.Dial(opts)
+	if err != nil {
+		return client, err
+	}
+
+	return &CustomClosingClient{
+		Client:      client,
+		statsCloser: clientScopeCloser,
+	}, nil
+}
+
+type CustomClosingClient struct {
+	client.Client
+	statsCloser io.Closer
+}
+
+func (c *CustomClosingClient) Close() {
+	c.Client.Close()
+
+	if c.statsCloser != nil {
+		c.statsCloser.Close()
+	}
+}
+
+func NewMetricsInterceptor(scope tally.Scope) interceptor.ClientInterceptor {
+	return &clientMetricsInterceptor{
+		scope: scope,
+	}
+}
+
+type clientMetricsInterceptor struct {
+	scope tally.Scope
+	// according to docs we should be embedding this for future changes
+	interceptor.ClientInterceptorBase
+}
+
+func (i *clientMetricsInterceptor) InterceptClient(
+	next interceptor.ClientOutboundInterceptor,
+) interceptor.ClientOutboundInterceptor {
+
+	return &clientMetricsOutboundInterceptor{
+		scope: i.scope,
+		ClientOutboundInterceptorBase: interceptor.ClientOutboundInterceptorBase{
+			Next: next,
+		},
+	}
+
+}
+
+type clientMetricsOutboundInterceptor struct {
+	scope tally.Scope
+	// according to docs we should be embedding this for future changes
+	interceptor.ClientOutboundInterceptorBase
+}
+
+func (i *clientMetricsOutboundInterceptor) ExecuteWorkflow(ctx context.Context, in *interceptor.ClientExecuteWorkflowInput) (client.WorkflowRun, error) {
+	s := i.scope.SubScope("execute_workflow")
+
+	timer := s.Timer("latency").Start()
+	defer timer.Stop()
+
+	run, err := i.ClientOutboundInterceptorBase.Next.ExecuteWorkflow(ctx, in)
+
+	if err != nil {
+		s.Counter("error").Inc(1)
+		return run, err
```

Comment:
nit: we could eliminate 2 return statements by wrapping the counter increment logic in an if-else block and do the same for other methods too? What do you think?

Reply:
it's a pretty marginal improvement and I don't like elses haha so I'm going to leave as is.

Reply:
Ahh okay!